### PR TITLE
Dependencies and settings update

### PR DESCRIPTION
**Pull request by kiroco12**
This PR updates the dependencies to their latest (and compatible) versions. These are not the final dependencies, so in a future the list can be changed.

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -49,30 +51,30 @@ lazy val server = (project in file("server"))
...
-      "com.vmunier"             %% "scalajs-scripts"  % v('scalajsscripts),
-      "com.lihaoyi"             %% "upickle"          % v('upickle),
-      "org.webjars"             %% "webjars-play"     % v('webjars),
-      "org.webjars"             % "bootstrap-sass"    % v('bootstrap),
-      "org.webjars"             % "highlightjs"       % v('highlightjs),
-      "com.tristanhunt"         %% "knockoff"         % v('knockoff),
-      "com.newrelic.agent.java" % "newrelic-agent"    % v('newrelic),
-      "commons-io"              % "commons-io"        % v('commonsio),
-      %%("freestyle"),
+      //"org.scala-exercises" %% "exercises-stdlib" % v('stdlib) xscalaExercises,
```
Comment (calvellido) [inline build.sbt]: Could we simply remove the unnecessary dependencies, even it they could come back in the future.

File: project/plugins.sbt [hunk 2]
```diff
@@ -4,21 +4,20 @@ logLevel := Level.Warn
...
+addSbtPlugin("com.eed3si9n"       % "sbt-buildinfo"            % "0.9.0")
+addSbtPlugin("com.47deg"          % "sbt-org-policies"         % "0.12.0-M3")
+addSbtPlugin("com.typesafe.play"  % "sbt-plugin"               % "2.8.0-M5")
+addSbtPlugin("org.portable-scala" % "sbt-scalajs-crossproject" % "0.6.1")
+addSbtPlugin("com.vmunier"        % "sbt-web-scalajs"          % "1.0.9-0.6")
+addSbtPlugin("com.typesafe.sbt"   % "sbt-gzip"                 % "1.0.2")
 
 // Scripted
-libraryDependencies += "org.scala-sbt" % "scripted-plugin" % sbtVersion.value
+libraryDependencies += "org.scala-sbt" %% "scripted-plugin" % sbtVersion.value
```
Comment (calvellido) [inline project/plugins.sbt]: Is this ok? :thinking:

Comment (kiroco12) [reply project/plugins.sbt]: Yes, needs the double '%' to work

File: client/src/main/scala/scripts/ExercisesJS.scala [hunk 3]
```diff
@@ -1,7 +1,5 @@
 /*
- *  scala-exercises
- *
- *  Copyright 2015-2017 47 Degrees, LLC. <http://www.47deg.com>
+ * Copyright 2016-2019 47 Degrees, LLC. <http://www.47deg.com>
```
Comment (vejeta) [inline client/src/main/scala/scripts/ExercisesJS.scala]: It is 2015 everywhere, why change it to 2016 here?

File: client/src/main/scala/utils/DomHandler.scala [hunk 4]
```diff
@@ -1,7 +1,5 @@
 /*
- *  scala-exercises
- *
- *  Copyright 2015-2017 47 Degrees, LLC. <http://www.47deg.com>
+ * Copyright 2016-2019 47 Degrees, LLC. <http://www.47deg.com>
```
Comment (vejeta) [inline client/src/main/scala/utils/DomHandler.scala]: It is 2015 everywhere, why change it to 2016 here?

File: client/src/main/scala/scripts/ExercisesJS.scala (hunk 3 above)
Comment (kiroco12) [reply client/src/main/scala/scripts/ExercisesJS.scala]: In build.sbt in the copyright part says 2015 but the `startDate` option says 2016. Which one is correct?

File: build.sbt [hunk 5]
```diff
@@ -137,43 +124,42 @@ lazy val definitions = (project in file("definitions"))
...
-      "org.scalariform" %% "scalariform"   % v('scalariform),
-      "org.scala-lang"  % "scala-compiler" % scalaVersion.value % "compile",
+      "org.scala-lang" % "scala-compiler" % scalaVersion.value % "compile",
       %%("github4s"),
       %%("cats-core") % "compile",
       %%("cats-laws") % "test",
       %%("scalatest") % "test"
-    )
+    ),
+    addSbtPlugin("org.scalariform" % "sbt-scalariform" % "1.8.3")
```
Comment (AdrianRaFo) [inline build.sbt]: TODO move to scalafmt

File: project/ProjectPlugin.scala [hunk 6]
```diff
@@ -83,37 +87,38 @@ object ProjectPlugin extends AutoPlugin {
         organizationEmail = "hello@47deg.com"
       ),
       orgLicenseSetting := ApacheLicense,
-      scalaVersion := "2.11.11",
+      scalaVersion := "2.12.10",
       scalaOrganization := "org.scala-lang",
-      crossScalaVersions := Seq("2.11.8"),
+      crossScalaVersions := Seq("2.12.10"),
```
Comment (AdrianRaFo) [inline project/ProjectPlugin.scala]: if we are going to just maintain the scala 2.12 version you don't need the crossCompile setting

**Comment by vejeta** [comment]
@kiroco12 Are the errors in Travis expected?

**Comment by kiroco12** [comment]
@vejeta yes, are expected. There isn't any code modifications so it won't compile

File: build.sbt [hunk 7]
```diff
@@ -191,23 +177,23 @@ lazy val `sbt-exercise` = (project in file("sbt-exercise"))
...
-  .enablePlugins(BuildInfoPlugin)
+// scripted plugin
+.enablePlugins(SbtPlugin)
+.settings(
+  scriptedLaunchOpts := {
+    scriptedLaunchOpts.value ++
+      Seq("-Xmx1024M", "-Dplugin.version=" + version.value)
+  },
+  scriptedBufferLog := false,
+//  // Publish definitions before running scripted
```
Comment (juanpedromoreno) [inline build.sbt]:
```suggestion
  // Publish definitions before running scripted
```

File: build.sbt [hunk 8]
```diff
@@ -137,43 +124,42 @@ lazy val definitions = (project in file("definitions"))
...
+    ),
+    addSbtPlugin("org.scalariform" % "sbt-scalariform" % "1.8.3")
   )
   .dependsOn(definitions, runtime)
 
 // Compiler plugin
 
 lazy val `sbt-exercise` = (project in file("sbt-exercise"))
-  .disablePlugins(CoursierPlugin)
+//
```
Comment (juanpedromoreno) [inline build.sbt]:
```suggestion

```

File: build.sbt [hunk 9]
```diff
@@ -16,23 +18,21 @@ lazy val `scala-exercises` = (project in file("."))
 
 // Purely functional core
 
-lazy val core = (crossProject in file("core"))
+lazy val core = (crossProject(JSPlatform, JVMPlatform) in file("core"))
```
Comment (AdrianRaFo) [inline build.sbt]: I think we just need JVMPlatform, aren't we?

Comment (kiroco12) [reply build.sbt]: The core project needs cross compilation because the client depends on the js core and the server needs the jvm core.

**Pull request merged**